### PR TITLE
feat: add did rpc decorations

### PR DESCRIPTION
**Pull request by weichweich**
## Fixes https://github.com/KILTprotocol/ticket/issues/1953

It adds type augmentation for the runtime APIs (callable with `api.call.didApi...`), ~but it has on hold the generation of types for the DID RPC stuff, as decoding is currently failing when fetching data using a Typescript script~.

## Checklist

- [x] Add definitions for RPC endpoints
- [x] Update polkadot package once its released with support for `AccountId33` as in https://github.com/polkadot-js/api/pull/5197/files

## How to test

After https://github.com/KILTprotocol/type-definitions/pull/33 is merged into this, you can spin up a local instance of polkadotJS apps, where you replace the KILT type definition with a locally published version of this package (e.g. with `yalc`). You should now be able to see the pubilc credential RPCs among the possible RPC methods, and also the runtime calls for both the DID and the public credentials runtime APIs.

One screenshot for the RPC and one for the runtime calls attached.

Screenshot 2022-08-11 at 16 24 16
Screenshot 2022-08-11 at 16 25 30

## How to integrate into the SDK

For usage with the SDK, the `typegen` package must be used, by first pulling in the metadata from an up-to-date chain that has these features (e.g. from `develop`), then updating `packages/augment-api/src/interfaces/extraDefs/definitions.ts` to use the `rpc`, `types`, and `runtime` definitions by pulling them in from this package, and the run `build:types`. The result should be an up-to-date typescript definition of all RPC and `state_call` stuff.

### CODE REVIEW

**Comment by ntn-x2** [comment]
We don't use this anymore in the SDK. The RPC info is populated directly from the runtime definition.

**Review by weichweich — COMMENTED** [review]
Would love to approve my on PR. But GitHub seems to be against that for what ever reason... 🙄

**Review by ntn-x2 — APPROVED** [review]
👌

**Comment by ntn-x2** [comment]
I have tested it with a local instance of the PolkadotJS Apps against both our latest develop and with Spiritnet and it seems to be working fine. To avoid any unnecessary risk with the official PolkadotJS Apps, I will leave this PR open and we will wait to merge it until we have started the release process for 1.8.0, at which point we will merge and release, and test the new types against Peregrine Staging.

**Comment by ntn-x2** [comment]
There is currently a limitation in how public keys are serialized in Substrate. They are ss58-encoded, and this means that there is need for some custom decoding logic on the client side. Commit https://github.com/KILTprotocol/type-definitions/pull/32/commits/11eb4f838e440853c11c7bd8edf83980c88548b3 fixes this by declaring the DID public verification keys as `Text`, which is the ss58-encoded representation of the public key. RPC consumers will have to call `decodeAddress(public_key)` to retrieve the byte representation of the key, if required.

Anyway, that was the last limitation for fully supporting the new RPC endpoints, hence we can now move on and test the new type definitions.

**Comment by ntn-x2** [comment]
This PR has been re-scoped to support the [1.7.2 release](https://github.com/KILTprotocol/kilt-node/pull/401), which adds support for the DID RPC endpoints, but does not include neither public credentials nor the new Ethereum account linking. Hence, those two features have been removed from this PR, and will be tracked in different ticekts. The goal of this PR is to make the type definitions ready for the next release.

File: src/types_10720.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import type {
+  RegistryTypes,
+} from "@polkadot/types/types"
+
+export const types10720: RegistryTypes = {
```
Comment (wischli) [inline src/types_10720.ts:5]: It's sufficient to just define DID related types here right?

**Review by wischli — APPROVED** [review]
LGTM! I have one minor question though.

File: src/types_10720.ts (hunk 1 above)
Comment (ntn-x2) [reply src/types_10720.ts:5]: Yes, since all the runtime types are taken from the metadata. We only need to declare types that are used for RPC endpoints.

**Pull request merged**